src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,101 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    //-----------------constants------------------------//
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int INITIAL_THRESHOLD = 12;
+    private static final int INCREASE_MULTIPLIER = 2;
+
+    // ----------------Node class-----------------------//
+    static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+    //----------------------fields------------------------//
+
+    private Node<K, V>[] map;
+    private int size;
+    private int threshold;
+    //---------------------constructor--------------------//
+
+    public MyHashMap() {
+        map = new Node[INITIAL_CAPACITY];
+        size = 0;
+        threshold = INITIAL_THRESHOLD;
+    }
+    //-------------------public methods-------------------//
 
     @Override
     public void put(K key, V value) {
-
+        if (map[getIndex(key)] == null) {
```

Comment:
better save the result of method getIndex(key) to thee local variable and use this variable where it is needed
@AleksandrPok FYI